broker/src/main/java/io/zeebe/broker/Broker.java

```diff
@@ -403,11 +404,12 @@ private AutoCloseable partitionsStep(
                     partitionIndexes.get(partitionId),
                     snapshotStoreSupplier,
                     createFactory(topologyManager, clusterCfg, atomix, managementRequestHandler),
-                    buildExporterRepository(brokerCfg),
-                    topologyManager);
+                    buildExporterRepository(brokerCfg));
             final PartitionTransitionImpl transitionBehavior =
                 new PartitionTransitionImpl(context, LEADER_STEPS, FOLLOWER_STEPS);
             final ZeebePartition zeebePartition = new ZeebePartition(context, transitionBehavior);
+            zeebePartition.addFailureListener(
```

Comment:
This should be called after `scheduleActor(zeebePartition)`. The reason is: it is executed with `actor.run`. If the actor is not started, it won't be executed.

Reply:
Done